pkg/build/controller/build_controller.go

```diff
@@ -130,6 +132,15 @@ func (bc *BuildController) HandleBuild(build *buildapi.Build) error {
 		return err
 	}
 
+	// Handle processing ValueFrom in build strategy environment variables
+	originalEnvVars, processedEnvVars, err := common.ProcessValueFrom(build, bc.KubeClient)
+	if err != nil {
+		build.Status.Phase = buildapi.BuildPhaseFailed
+		build.Status.Reason = buildapi.StatusReasonInvalidValueFrom
+		return err
+	}
+	*originalEnvVars = *processedEnvVars
```

Comment:
this should be done at the same point where we apply env from the defaults/overrides.

in fact this processing should be done after that, so it can take into account default/override values that use a valuefrom reference.